### ISSUE TITLE
aws breaking changes 20231102

### ISSUE THREAD

**Issue by Nugine** [Nugine/s3s#100]
+ https://github.com/awslabs/aws-sdk-rust/blob/main/CHANGELOG.md#november-2nd-2023
+ https://github.com/awslabs/smithy-rs/blob/main/CHANGELOG.md#november-1st-2023
+ https://github.com/Nugine/s3s/pull/99

#100, #107, #111 

+ [ ] update to the latest smithy model and adjust codegen
+ [ ] update to aws sdk 1.0
+ [ ] update to hyper 1.0

**Comment by Nugine** [comment]
I'm busy recently. If someone's work is blocked by this issue, please let me know. I'll try to prioritize this issue.

**Comment by Nugine** [comment]
It's more difficult than I thought. Waiting for hyper 1.0 in aws-sdk-rust.

**Comment by edaniels** [comment]
This just came up for me as we're trying to upgrade our AWS crates to latest

**Comment by edaniels** [comment]
Thank you!

**Comment by Nugine** [comment]
The next release v0.9.0 is scheduled for Mar 9th.
This version contains lots of breaking changes introduced by the latest smithy model. You can test it via git dependency before the release date. Report any problem :)
If we encounter any major issue, the release will be postponed until we fix it.
